cloudformation/tests/test_cluster_custom_resource.py

```diff
@@ -0,0 +1,246 @@
+"""Test cases for Cluster CloudFormation Custom Resource."""
+import random
+import string
+
+import botocore
+import pytest
+import urllib3
+import yaml
+from assertpy import assert_that
+from conftest import cfn_stack_generator
+
+CLUSTER_TEMPLATE = "../custom_resource/cluster.yaml"
+TEST_CLUSTER = "test_cluster.yaml"
+
+
+# The following functions late-bind the library as it is only used locally
+def _list_clusters(cluster_name):
+    """List clusters and filter by name."""
+    import pcluster.lib as pcluster
+
+    return pcluster.list_clusters(query=f"clusters[?clusterName=='{cluster_name}']|[0]")
+
+
+def _describe_cluster(cluster_name):
+    """Late bind library and describe cluster."""
+    import pcluster.lib as pcluster
+
+    return pcluster.describe_cluster(cluster_name=cluster_name)
+
+
+def _delete_cluster(cluster_name):
+    """Late bind library and delete cluster."""
+    import pcluster.lib as pcluster
+
+    return pcluster.delete_cluster(cluster_name=cluster_name)
+
+
+def _random_id():
```

Comment:
nitpick, non-blocking, why aren't we using `_random_str()` from `conftest.py` again?